internal/pkg/source/testdata/src/analyzertest/sourcetest/identification.go

```diff
@@ -20,6 +20,34 @@ type Alias = Source
 // type definition where the underlying type is a Source
 type Definition Source
 
+type FuncField struct {
+	flipCoin func() bool
+	producer func() Source
+	ptr      func() *Source
+}
+
+func TestFunctionFields() {
```

Comment:
I'm not sure which cases are covered by these tests.

Does this cover `*types.Signature`? (It doesn't seem to be the case, to me.)

Reply:
I looked at the docs for `Signature` and it looks like it means `Function`:
```go
// A Signature represents a (non-builtin) function or method type.
// The receiver is ignored when comparing signatures for identity.
```
Maybe add a line where you assign a `func` to a variable? Would that trigger the `*types.Signature` case?

Reply:
This also raises the question of how `builtins` are represented, specifically what is their type? :thinking:

Reply:
> This also raises the question of how builtins are represented, specifically what is their type? 🤔

There is a `types.Builtin`, but it doesn't implement `types.Type`.  So that can't be returned by the `ssa.Builtin.Type()`.  I think this might be the same issue with `types.Func`, which also does not implement `types.Type`.

I'm starting to appreciate the various comments around the SSA library about the difficulty in mapping between these packages.